server/server_test.go

```diff
@@ -143,6 +143,16 @@ func TestVersionMatchesTag(t *testing.T) {
 	if VERSION != tag[1:] {
 		t.Fatalf("Version (%s) does not match tag (%s)", VERSION, tag[1:])
 	}
+	// Check that the version dynamically set via ldflags matches the version
+	// from the server previous to releasing.
+	if serverVersion == _EMPTY_ {
+		t.Fatal("Version missing in ldflags")
+	}
+	// Unlike VERSION constant, this one is prefixed with a 'v'
+	// since it is the same as the git tag.
+	if serverVersion != _EMPTY_ && VERSION != serverVersion[1:] {
```

Comment:
A slight improvement to confirm there is a `v` prefix would be to do: `"v" + VERSION != serverVersion`

Reply:
good idea, should check that `serverVersion` does include the `v` prefix